Library/EnvelopeFactory.ts

```diff
@@ -0,0 +1,329 @@
+import Telemetry = require("./Telemetry")
+import DependencyTelemetry = require("./DependencyTelemetry")
+import EventTelemetry = require("./EventTelemetry")
+import TraceTelemetry = require("./TraceTelemetry")
+import ExceptionTelemetry = require("./ExceptionTelemetry")
+import RequestTelemetry = require("./RequestTelemetry")
+import MetricTelemetry = require("./MetricTelemetry")
+
+import Contracts = require("../Declarations/Contracts")
+import Util = require("./Util")
+import Config = require("./Config");
+import Context = require("./Context");
+import { CorrelationContextManager } from "../AutoCollection/CorrelationContextManager";
+
+/**
+ * Manages the logic of creating envelopes from Telemetry objects
+ */
+class EnvelopeFactory {
+
+    /**
+     * Creates envelope ready to be sent by Channel
+     * @param telemetry Telemetry data
+     * @param telemetryType Type of telemetry
+     * @param commonProperties Bag of custom common properties to be added to the envelope
+     * @param context Client context
+     * @param config Client configuration
+     */
+    public static createEnvelope(
+        telemetry: Telemetry,
+        telemetryType: string,
+        commonProperties?: { [key: string]: string; },
+        context?: Context,
+        config?: Config): Contracts.Envelope {
+
+        var data:
```

Comment:
Is it necessary to enumerate all event types here?

Reply:
Shortened.